x/emissions/keeper/inference_synthesis/weight.go

```diff
@@ -140,46 +162,50 @@ func (p *SynthPalette) CalcWeightedInference(weights RegretInformedWeights) (Inf
 			}
 		}
 	} else {
-		for _, inferer := range p.Inferers {
-			if _, ok := p.InferenceByWorker[inferer]; !ok {
-				p.Logger.Debug(fmt.Sprintf("Cannot find inferer in InferenceByWorker in CalcWeightedInference %v", inferer))
+		for _, inferer := range inferers {
+			inferenceByWorker, exists := workerToInference[inferer]
+			if !exists {
+				logger.Debug(fmt.Sprintf("Cannot find inferer in InferenceByWorker in CalcWeightedInference %v", inferer))
 				continue
 			}
-			if _, ok := weights.inferers[inferer]; !ok {
-				p.Logger.Debug(fmt.Sprintf("Cannot find inferer in weights.inferers in CalcWeightedInference %v", inferer))
+			infererWeight, exists := weights.inferers[inferer]
+			if !exists {
+				logger.Debug(fmt.Sprintf("Cannot find inferer in weights.inferers in CalcWeightedInference %v", inferer))
 				continue
 			}
-			if _, ok := p.InfererRegrets[inferer]; !ok {
-				p.Logger.Debug(fmt.Sprintf("Cannot find inferer in InfererRegrets in CalcWeightedInference %v", inferer))
+			if _, exists := infererToRegret[inferer]; !exists {
+				logger.Debug(fmt.Sprintf("Cannot find inferer in InfererRegrets in CalcWeightedInference %v", inferer))
 				continue
 			}
-			runningUnnormalizedI_i, sumWeights, err = AccumulateWeights(
-				p.InferenceByWorker[inferer],
-				weights.inferers[inferer],
-				p.AllInferersAreNew,
+			runningUnnormalizedI_i, sumWeights, err = accumulateWeights(
+				*inferenceByWorker,
+				infererWeight,
+				allInferersAreNew,
 				runningUnnormalizedI_i,
 				sumWeights,
 			)
 			if err != nil {
 				return InferenceValue{}, errorsmod.Wrapf(err, "Error accumulating weight of inferer")
 			}
 		}
-		for _, forecaster := range p.Forecasters {
-			if _, ok := p.ForecastImpliedInferenceByWorker[forecaster]; !ok {
-				p.Logger.Debug(fmt.Sprintf("Cannot find forecaster in ForecastImpliedInferenceByWorker in CalcWeightedInference %v", forecaster))
+		for _, forecaster := range forecasters {
+			workerForecastImpliedInference, exists := forecasterToForecastImpliedInference[forecaster]
+			if !exists {
+				logger.Debug(fmt.Sprintf("Cannot find forecaster in ForecastImpliedInferenceByWorker in CalcWeightedInference %v", forecaster))
 				continue
 			}
-			if _, ok := weights.forecasters[forecaster]; !ok {
-				p.Logger.Debug(fmt.Sprintf("Cannot find forecaster in weights.forecasters in CalcWeightedInference %v", forecaster))
+			forecasterWeight, exists := weights.forecasters[forecaster]
+			if !exists {
+				logger.Debug(fmt.Sprintf("Cannot find forecaster in weights.forecasters in CalcWeightedInference %v", forecaster))
 				continue
 			}
-			if _, ok := p.ForecasterRegrets[forecaster]; !ok {
-				p.Logger.Debug(fmt.Sprintf("Cannot find forecaster in ForecasterRegrets in CalcWeightedInference %v", forecaster))
+			if _, exists := forecasterToRegret[forecaster]; !exists {
```

Comment:
I believe resident Go sage @spooktheducks frowns upon this?

Reply:
will fix